BlogTemplate/Pages/Account/ConfirmEmail.cshtml.cs

```diff
@@ -28,13 +28,13 @@ public ConfirmEmailModel(UserManager<ApplicationUser> userManager)
             var user = await _userManager.FindByIdAsync(userId);
             if (user == null)
             {
-                throw new ApplicationException($"Unable to load user with ID '{userId}'.");
+                throw new FieldAccessException($"Unable to load user with ID '{userId}'.");

```

Comment:
@Eilon I think he meant using `return BadRequest();` instead of throwing an exception.  Basically, getting an HTTP 400 instead of 500 in situations where the request contained invalid data.  I think this case is also appropriate since the username is provided in the GET request, so an invalid ID is the requestor's fault.

But in cases below like ChangePassword where it does `var user = await _userManager.GetUserAsync(User);` I still think throwing (HTTP500) is appropriate - i.e. you're logged in (so the request is probably valid so far as we know), but somehow mapping that through the _userManager went wrong.